### PR TITLE
Rebindable communication(OOC, Say, Me)

### PR DESCRIPTION
**Pull request by Couls**
## About The Pull Request
I gotta get these in before november
OOC, Say and Me is now rebindable

## Why It's Good For The Game
No more accidental ick ock

## Changelog
:cl:
add: You can now rebind OOC, Say and Me
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
Note for the maintainers reviewing this, I have the savefile update commit stashed so please let me know before hand when this is about to be merged so I can add the savefile update

File: code/datums/keybinding/communication.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/datum/keybinding/client/communication/say
+	hotkey_keys = list("T")
+	name = "Say"
+	full_name = "IC Say"
+	keybind_signal = COMSIG_KB_CLIENT_SAY_DOWN
+
+/datum/keybinding/client/communication/ooc
+	hotkey_keys = list("O")
+	name = "OOC"
+	full_name = "Out Of Character Say(OOC)"
```
Comment (Jared-Fogle) [inline code/datums/keybinding/communication.dm:13]:
```suggestion
	full_name = "Out Of Character Say (OOC)"
```

File: code/datums/keybinding/communication.dm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	keybind_signal = COMSIG_KB_CLIENT_OOC_DOWN
+
+/datum/keybinding/client/communication/me
+	hotkey_keys = list("M")
+	name = "Me"
+	full_name = "Custom Emote(/Me)"
```
Comment (Jared-Fogle) [inline code/datums/keybinding/communication.dm:19]:
```suggestion
	full_name = "Custom Emote (/Me)"
```

**Comment by Jared-Fogle** [comment]
> Note for the maintainers reviewing this, I have the savefile update commit stashed so please let me know before hand when this is about to be merged so I can add the savefile update

Just do it at the same time?

**Comment by Couls** [comment]
but what if one of the maints wants a testmerge? 😳

**Comment by Couls** [comment]
savefile update won't be necessary once #54050 is merged

**Comment by Rohesie** [comment]
Actually this doesn't look like it needs a TM.
Have you binded OOC to several keys locally, and see it properly unbound from the previous and into the new one correctly?

**Comment by kriskog** [comment]
Does this implementation get rid of the lag that was the reason for reverting these to macros last time around?

**Comment by Couls** [comment]
Yes both the questions above, there shouldn’t be massive lag this time around because the binds still use macros in the code and the communication binds should be properly unbound because all macros are erased and then reapplied every time they’re changed or updated(set_macros is always called before update_special_keys)

**Comment by Couls** [comment]
Done and tested, changing the key changes the bind accordingly

**Pull request merged**